### PR TITLE
Add option to force reading relTh from ini 

### PR DESCRIPTION
**Pull request by awirb**
* update log so that it tells you where the release thickness is read from in addition to its value 
* update docu with a note on this 
* update glossary - more info required?
* update shapefiles 
* update warning
* also read entTh from shape file 
* hEnt -> entTh 
* close issue #494  here

### CODE REVIEW

**Review by fso42 — APPROVED** [review]

File: avaframe/in2Trans/shpConversion.py [hunk 1]
```diff
@@ -57,11 +57,11 @@ def SHP2Array(infile, defname=None):
     infile = pathlib.Path(infile)
     # set defaults for variables
     layername = None
-    d0 = None
+    relTh = None
     rho = None
     sks = None
     iso = None
-    d01 = []
+    relTh1 = []
```
Comment (fso42) [inline avaframe/in2Trans/shpConversion.py:64]: What is this needed for?

File: docs/moduleCom1DFA.rst [hunk 2]
```diff
@@ -43,6 +43,9 @@ In the directory ``Inputs``, the following files are required:
 * release area scenario as shapefile (in Inputs/REL); multiple features are possible
   -> the release area name should not contain an underscore, if so '_AF' is added
 
+.. Note:: Release thickness can be specified in the shape file by adding an attribute called `relTh` or via the
```
Comment (fso42) [inline docs/moduleCom1DFA.rst:46]: Please add:
For backwards compatibility, the attribute 'd0' also works, but we suggest to use relTh in new projects.

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/in2Trans/shpConversion.py (hunk 1 above)
Comment (awirb) [reply avaframe/in2Trans/shpConversion.py]: this is the list where the thickness values of the individual features are appended i renamed it to thicknessList and moved it to the respective list initialisation for the name

File: docs/moduleCom1DFA.rst (hunk 2 above)
Comment (awirb) [reply docs/moduleCom1DFA.rst]: Added

File: avaframe/com1DFA/com1DFA.py [hunk 3]
```diff
@@ -220,7 +223,7 @@ def com1DFACore(cfg, avaDir, cuSimName, inputSimFiles, outDir, relThField):
     return particlesList, fieldsList, Tsave, dem, reportDict, cfg
 
 
-def prepareRelase(cfg, rel, inputSimLines):
+def prepareReleaseEntrainment(cfg, rel, inputSimLines):
```
[the remark this answers is not in the document]
Comment (fso42) [reply avaframe/com1DFA/com1DFA.py:227]: inputSimLines is missing from the docstring

File: avaframe/com1DFA/com1DFA.py [hunk 4]
```diff
@@ -253,14 +256,14 @@ def prepareRelase(cfg, rel, inputSimLines):
...
+    releaseLine['thicknessSource'] = [''] * len(releaseLine['thickness'])
+    for k in range(len(releaseLine['thickness'])):
+        if cfg['GENERAL'].getboolean('useRelThFromIni') or releaseLine['thickness'][k] == 'None':
+            releaseLine['thickness'][k] = cfg['GENERAL'].getfloat('relTh')
+            releaseLine['thicknessSource'][k] = 'ini file'
         else:
-            releaseLine['relTh'][k] = float(releaseLine['relTh'][k])
-            releaseLine['relThSource'][k] = 'shp file'
+            releaseLine['thickness'][k] = float(releaseLine['thickness'][k])
+            releaseLine['thicknessSource'][k] = 'shp file'
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py]: This (and the other similiar ones) can be simplified (at least I think so):
- if use from ini is set, all release lines get the same value, ie. all thicknesses and source can be set at once
- if shp file is set, all sources can be set at once,
- therefore the only loop that is needed is that for the conversion to float

**Review by fso42 — CHANGES_REQUESTED** [review]

File: docs/glossary.rst [hunk 5]
```diff
@@ -14,13 +14,13 @@ Glossary
...
+      consistently below 10 degrees.
 
    beta angle
-      Angle from the topmost point of a profile to the :term:`beta point`. 
+      Angle from the topmost point of a profile to the :term:`beta point`.
 
-   release thickness [*relTh*, *d0*]
-      TODO: which symbol 
+   release thickness [*relTh*]
+      TODO: which symbol
```
Comment (fso42) [inline docs/glossary.rst:23]:
```suggestion
   release thickness [*relTh*]
      ```

Also add the same for entrainment thickness

Comment (awirb) [reply docs/glossary.rst:23]: done

File: avaframe/com1DFA/com1DFA.py [hunk 6]
```diff
@@ -244,20 +250,17 @@ def prepareRelase(cfg, rel, inputSimLines):
...
         the suffix _AF will be added for the simulation name')
-    releaseLine = inputSimLines['releaseLine']
-    for k in range(len(releaseLine['d0'])):
-        if releaseLine['d0'][k] == 'None':
-            releaseLine['d0'][k] = cfg['GENERAL'].getfloat('relTh')
-        else:
-            releaseLine['d0'][k] = float(releaseLine['d0'][k])
+
+    # set release thickness
+    releaseLine = setThickness(cfg, inputSimLines['releaseLine'], 'useRelThFromIni', 'relTh')
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:263]: useRelThFromIni is hardcoded now? Is that wanted?

File: avaframe/com1DFA/com1DFA.py [hunk 7]
```diff
@@ -266,21 +269,50 @@ def prepareRelase(cfg, rel, inputSimLines):
...
 
 
+def setThickness(cfg, lineTh, flagTh, typeTh):
+    """ set thickness in line dictionary for release area, entrainment area
+
+    Parameters
+    -----------
+    lineTh: dict
+        dictionary with info on line (e.g. release area line)
+    flagTh: bool
```
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:293]: See issue #86 : we are trying not to use flagSOMETHING, so maybe change this to useThFromIni

Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:293]: changed to useThFromIni

File: avaframe/com1DFA/com1DFA.py (hunk 6 above)
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:263]: yes, this is the flag that is passed to the function setThickness (which is new - to reduce duplication of code as was almost the same code block for release, secrel and entrainment)

**Pull request merged**